src/snowflake/connector/cache.py

```diff
@@ -595,8 +614,9 @@ def clear(self) -> None:
 
     def __getstate__(self) -> dict:
         state = self.__dict__.copy()
-        del state["_lock"]
-        del state["_file_lock"]
+        for k in list(state.keys()):
+            if k not in SFDictFileCache._ATTRIBUTES_TO_PICKLE:
+                del state[k]
```

Comment:
I wonder if we could just do: `state = {k: v for k, v in state.items() if k in SFDictFileCache._ATTRIBUTES_TO_PICKLE}`

Reply:
thanks for the suggestion, updated implementation